src/execution_plan/execution_plan.c

```diff
@@ -658,18 +658,35 @@ ExecutionPlan* NewExecutionPlan(RedisModuleCtx *ctx, AST **ast, ResultSet *resul
     return plan;
 }
 
-void _ExecutionPlan_Print(const OpBase *op, RedisModuleCtx *ctx) {
+void _ExecutionPlan_Print(const OpBase *op, RedisModuleCtx *ctx, char *buffer, int ident, int *op_count) {
     if(!op) return;
 
-    RedisModule_ReplyWithArray(ctx, op->childCount + 1);
-    RedisModule_ReplyWithSimpleString(ctx, op->name);
+    *op_count += 1; // account for current operation.
+
+    // Construct operation string representation.
+    int offset = 0;
+    for(int i = 0; i < ident; i++) offset += sprintf(buffer+offset, "    ");
+    int len = sprintf(buffer+offset, "%s", op->name);
```

Comment:
We can build our indentation more simply with field width specifiers:
```
int len = sprintf(buffer, "%*s%s", ident, "", op->name);
```
This replaces the need for all the `offset` logic and additional `sprintf` calls. (Assuming `ident` is incremented by 4 on 674).